packages/react/src/internals/getArrayQueryKey.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * To allow easy interactions with groups of related queries, such as
+ * invalidating all queries of a router, we use an array as the path when
+ * storing in tanstack query. This function converts from the `.` separated
+ * path passed around internally by both the legacy and proxy implementation.
+ * https://github.com/trpc/trpc/issues/2611
+ */
+export function getArrayQueryKey(
+  queryKey: string | [string] | [string, ...unknown[]] | unknown[],
+): [string[]] | [string[], ...unknown[]] | [] {
+  const queryKeyArrayed = Array.isArray(queryKey) ? queryKey : [queryKey];
+  const [path, ...input] = queryKeyArrayed;
+
+  const arrayPath = path ? path.split('.') : [];
```

Comment:
Heads-up: argument types technically allow an array in the first position (via `unknown[]`), but an array can't be `.split`.